src/docs/sphinx/dev_guide/updating_tpls.rst

```diff
@@ -0,0 +1,192 @@
+traditional sense. To update one of these packages in Axom, simply go into
+its directory in Axom and check out a new version. If a version is intended
+to be changed in the Axom repo, make the version change on a branch and 
+submit a GitHub pull request as you would do for other software changes.
+More info on :ref:`toolkitbuild-label`.
+
+Built-in TPLs
+-------------
+
+Axom several lightweight header-only libraries that we use internally and
+expose for downstream customers to use if they wish.
+
+  * `CLI11 <https://github.com/CLIUtils/CLI11>`_, is a command line parser
+    for C++11 and beyond that provides a rich feature set with a simple and
+    intuitive interface.
+  * `fmt <https://github.com/fmtlib/fmt>`_, is an open-source formatting
+    library providing a fast and safe alternative to C stdio and C++ iostreams.
+  * `sol <https://github.com/ThePhD/sol2>`_, is a C++ library binding to Lua.
+  * `Sparsehash <https://github.com/sparsehash/sparsehash>`_, contains several
+    hash-map implementations.
+
+.. note:: Axom patches all built-in TPLs to be under the `axom` namespace.
+   This is to prevent symbol collisions with other projects, either our
+   dependencies or downstream customers who wish their own versions.  For
+   example, `fmt::format("foo")` is `axom::fmt::format("foo")`.
+
+They can be found in our repository here: `src/thirdparty`. The basic 
+instructions on how to update a built-in TPL are as follows:
+
+#. Download the new release and override the source that is already there.
+   This can often involve removing no-longer needed files but most of the
+   current ones are a single header file.
+
+#. Review and apply the existing patch files. More than likely, you will not
+   be able to directly apply the patch but it will give you the general idea
+   on what needs to be applied.  For example, the namespace update mentioned above.
+
+#. Ensure that the build and tests still pass. More info on :ref:`testing-label`.
+
+#. Follow the normal pull request work flow. More info on :ref:`pullrequest-label`.
+
+
+TPLs (system-level and external libraries)
+------------------------------------------
+
+The following instructions describe how to install local copies of Axom
+TPLs on Livermore Computing (LC) platforms. Typically, this process is 
+followed when you want to build and test Axom against new versions of TPLs
+to make sure everything works properly before deploying a new TPL set for
+other Axom developers to use, to use in Axom CI testing, etc.
+
+#. **Working on a local branch.** 
+   Make a local clone of the Axom repo and create a branch to work on. Working
+   on a separate branch is a good practice to avoid propagating mis-steps
+   to other users and/or developers.
+
+#. **Changing versions of system packages or other TPLs.**
+   To change a version of a system package, which applies to an LC platforms 
+   or a Docker container image we use for CI testing on GitHub, go into
```

Comment:
It may be better to pull out the comments related to containers and GitHub/azure testing into their own section so it's clear what needs to be done to update each.

Reply:
There is a lot of overlap between the two.  I think if we break them into two sections it will be a lot of duplication. Azure has its own step down below that is separate from LC steps.  I will clarify which parts are which though.